app/src/main/java/ca/bc/gov/vaxcheck/MainActivity.kt

```diff
@@ -15,8 +24,156 @@ import dagger.hilt.android.AndroidEntryPoint
 @AndroidEntryPoint
 class MainActivity : AppCompatActivity() {
 
+    private val TAG = MainActivity::class.java.canonicalName
+
+    private lateinit var appUpdateManager: AppUpdateManager
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
+
+        checkAppUpdate()
```

Comment:
Can we move this entire implementation to the DashBoard Fragment?

Reply:
For flexible updates, there is a listener, which is active and listening for the background updates download. 

After the download completes this will show a snack bar notification to install the updates. 

If we move this code to fragment we have to keep the listener registered irrespective of fragment lifecycle and this will cause a memory leak.